salt/engines/napalm_syslog.py

```diff
@@ -0,0 +1,228 @@
+
+An engine that takes syslog messages structured in
+[OpenConfig](http://www.openconfig.net/) or IETF format
+and fires Salt events.
+
+As there can be many messages pushed into the event bus,
+the user is able to filter based on the object structure.
+
+Requirements
+------------
+
+- [napalm-logs](https://github.com/napalm-automation/napalm-logs)
+
+This engine transfers objects from the napalm-logs library
+into the event bus. The top dictionary has the following keys:
+
+- ``ip``
+- ``host``
+- ``timestamp``
+- ``os``: the network OS identified
+- ``model_name``: the OpenConfig or IETF model name
+- ``error`: the error name (consult the documentation)
+- ``message_details``: details extracted from the syslog message
+- ``open_config``: the OpenConfig model
+
+The napalm-logs transfers the messages via widely used transport
+mechanisms such as: ZeroMQ (default), Kafka, etc.
+
+The user can select the right transport using the ``transport``
+option in the configuration. The
+
+:configuration: Example configuration
+
+    .. code-block:: yaml
+
+        engines:
+          - napalm_syslog:
+                transport: zmq
+                address: 1.2.3.4
+                port: 49018
+
+Output object example:
+
+.. code-block:: json
+
+    {
+      "error": "BGP_PREFIX_THRESH_EXCEEDED",
+      "ip": "127.0.0.1",
+      "host": "re0.edge01.bjm01",
+      "message_details": {
+        "processId": "2902",
+        "error": "BGP_PREFIX_THRESH_EXCEEDED",
+        "pri": "149",
+        "processName": "rpd",
+        "host": "re0.edge01.bjm01",
+        "time": "12:45:19",
+        "date": "Mar 30",
+        "message": "1.2.3.4 (External AS 15169): Configured maximum prefix-limit threshold(160) exceeded for inet-unicast nlri: 181 (instance master)"
+      },
+      "model_name": "openconfig_bgp",
+      "open_config": {
+        "bgp": {
+          "neighbors": {
+            "neighbor": {
+              "1.2.3.4": {
+                "neighbor-address": "1.2.3.4",
+                "state": {
+                  "peer-as": 15169
+                },
+                "afi-safis": {
+                  "afi-safi": {
+                    "inet": {
+                      "state": {
+                        "prefixes": {
+                          "received": 181
+                        }
+                      },
+                      "ipv4-unicast": {
+                        "prefix-limit": {
+                          "state": {
+                            "max-prefixes": 160
+                          }
+                        }
+                      },
+                      "afi-safi-name": "inet"
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      },
+      "os": "junos",
+      "timestamp": "1490877919"
+    }
+
+
+'''
+from __future__ import absolute_import
+
+# Import python stdlib
+import json
+import logging
+
+# Import third party libraries
+import zmq
```

Comment:
No problem - added in https://github.com/saltstack/salt/pull/40458/commits/75c64bfe453d86f2f19a41fac5d2ccc49a40739b.